src/python/pants/engine/mapper.py

```diff
@@ -70,7 +69,7 @@ def parse(cls, filepath, filecontent, parser):
         raise DuplicateNameError('An object already exists at {!r} with name {!r}: {!r}.  Cannot '
                                  'map {!r}'.format(filepath, name, objects_by_name[name], obj))
       objects_by_name[name] = obj
-    return cls(filepath, OrderedDict(sorted(objects_by_name.items())))
```

Comment:
Dicts are ordered by insertion order in Python 3.6+ so using `OrderedDict` is redundant.

Reply:
Is insertion order deterministic here?

Reply:
Yes. In Python 3.6+, `OrderedDict` and `dict` both use deterministic insertion order. The `sored()` is what ensures that we always use the same order when inserting into the dict.